File: pset6/readability/readability.py
from cs50 import get_string

def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    index = 0.0588 * (letters/words*100) - 0.296 * (sentences/words*100) - 15.8
    grade = round(index)
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")

def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters = letters + 1
    return letters

def count_words(text):
    words = 1
    for i in range(len(text)):
        if text[i] == ' ':
            words = words + 1
    return words

def count_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            sentences = sentences + 1
    return sentences

main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    float index = 0.0588 * ((float)letters/(float)words*100) - 0.296 * ((float)sentences/(float)words*100) - 15.8;
    int grade = roundf(index);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n",grade);
    }

}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n  = strlen(text); i < n; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
    }
   return letters;
}

int count_words(string text)
{
    int words = 1;
    for (int i = 0, n  = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        words++;
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n  = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        sentences++;
    }
    return sentences;
}